codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -233,35 +232,32 @@ void onUnknownFrame0(ChannelHandlerContext ctx, byte frameType, int streamId, Ht
         @Override
         public int onDataRead(final ChannelHandlerContext ctx, int streamId, ByteBuf data,
                 int padding, boolean endOfStream) throws Http2Exception {
-            // Check if we received a data frame for a stream which is half-closed
             Http2Stream stream = connection.requireStream(streamId);
```

Comment:
@Scottmitch @buchgr Hmm ... this is actually a little tricky. [section 5.1](https://tools.ietf.org/html/draft-ietf-httpbis-http2-17#section-5.1):
> An endpoint MUST NOT send frames other than PRIORITY on a closed
      stream.  An endpoint that receives any frame other than PRIORITY
      after receiving a RST_STREAM MUST treat that as a stream error
      (Section 5.4.2) of type STREAM_CLOSED.  Similarly, an endpoint
      that receives any frames after receiving a frame with the
      END_STREAM flag set MUST treat that as a connection error
      (Section 5.4.1) of type STREAM_CLOSED, unless the frame is
      permitted as described below.

So judging by that text, it depends on how the stream was closed as to whether it should be a connection error or a stream error.

Also WRT flow control (same section):
> Flow controlled frames (i.e., DATA) received after sending
      RST_STREAM are counted toward the connection flow control window.
      Even though these frames might be ignored, because they are sent
      before the sender receives the RST_STREAM, the sender will
      consider the frames to count against the flow control window.

So if our endpoint terminated the stream, we still have to apply flow control to any received DATA frames.